### PR TITLE
[RLlib] Change `rllib train` default command line arg values.

### PR DESCRIPTION
**Pull request by sven1977**
Change `rllib train` default command line arg values.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/train.py [hunk 1]
```diff
@@ -43,20 +47,20 @@ def create_parser(parser_creator=None):
...
         "--ray-address",
-        default=None,
+        default="auto",
         type=str,
         help="Connect to an existing Ray cluster at this address instead "
         "of starting a new one.",
     )
     parser.add_argument(
-        "--ray-ui", action="store_true", help="Whether to enable the Ray web UI."
+        "--no-ray-ui", action="store_true", help="Disable Ray UI from being used."
```
Comment (rkooo567) [inline rllib/train.py:56]: Should we create a deprecation when the old option is used?

**Review by rkooo567 — APPROVED** [review]
LGTM. But I might be waiting for other rllib experts to also approve just in case!

**Review by rkooo567 — APPROVED** [review]

File: rllib/train.py [hunk 2]
```diff
@@ -186,9 +190,9 @@ def run(args, parser):
         }
 
     # Ray UI.
-    if args.no_ray_ui:
-        deprecation_warning(old="--no-ray-ui", new="--ray-ui", error=False)
-        args.ray_ui = False
+    if args.ray_ui:
```
Comment (sven1977) [inline rllib/train.py:193]: Here. @rkooo567 .

File: rllib/train.py [hunk 3]
```diff
@@ -257,14 +261,30 @@ def patch_path(path):
...
+        # Try connecting to given address (by default, this is "auto").
+        try:
+            ray.init(
+                include_dashboard=not args.no_ray_ui,
+                address=args.ray_address,
+                object_store_memory=args.ray_object_store_memory,
+                num_cpus=args.ray_num_cpus,
+                num_gpus=args.ray_num_gpus,
+                local_mode=args.local_mode,
+            )
```
Comment (ericl) [inline rllib/train.py:273]: Does this print anything bad to the console on failure?

File: rllib/train.py [hunk 4]
```diff
@@ -257,14 +261,30 @@ def patch_path(path):
...
+        # Last resort: Try running in a new ray cluster (address=None).
+        except ConnectionError:
+            logger.warning(
+                "Connecting to ray-address={} not successful! Trying to run in a "
+                "new ray cluster."
+            )
```
Comment (ericl) [inline rllib/train.py:279]: Hmm, this will be the common case. I don't think we want to make all laptop users see this warning.

**Comment by ericl** [comment]
Instead of doing this only for RLlib train, how about having the record_library_usage() call touch /tmp/ray/$library_used? Then any other Ray instance on the machine will pick up the library usage. Cc @rkooo567

**Comment by rkooo567** [comment]
Doesn't that mean if the machine uses the library once, all future instance is considered as they have the library usage?

**Comment by ericl** [comment]
Yes, that's the whole point, to make this cross Ray session. Let's do that instead of this PR, its a much more general and correct fix.

**Comment by ericl** [comment]
Superceded by https://github.com/ray-project/ray/pull/25842

**Pull request closed without merging**